src/steps/containers.rs

```diff
@@ -77,9 +76,22 @@ fn list_containers(crt: &Path, ignored_containers: Option<&Vec<String>>) -> Resu
         "Querying '{} image ls --format \"{{{{.Repository}}}}:{{{{.Tag}}}}/{{{{.ID}}}}\"' for containers",
         crt.display()
     );
-    let output = Command::new(crt)
-        .args(["image", "ls", "--format", "{{.Repository}}:{{.Tag}} {{.ID}}"])
-        .output_checked_utf8()?;
+
+    let sudo = if ctx.config().containers_use_sudo() {
+        Some(ctx.require_sudo()?)
+    } else {
+        None
+    };
+
+    let output = if let Some(sudo) = sudo {
```

Comment:
Just an internal tracker, there was a reason to use `Command::new` here, but it's impossible to do that with sudo :)
```suggestion
    // TODO: This should run even when dry-running. Blocked by #1227.
    let output = if let Some(sudo) = sudo {
```